api/src/mutations/drop.rs

```diff
@@ -240,6 +243,157 @@ impl Mutation {
             drop: Drop::new(drop_model, collection_model),
         })
     }
+
+    /// Res
+    ///
+    /// # Errors
+    /// This function fails if ...
+    pub async fn edit_drop(
+        &self,
+        ctx: &Context<'_>,
+        input: EditDropInput,
+    ) -> Result<CreateDropPayload> {
+        let EditDropInput {
+            drop,
+            price,
+            start_time,
+            end_time,
+            seller_fee_basis_points,
+            metadata_json,
+        } = input;
+
+        if price.is_none()
+            && start_time.is_none()
+            && end_time.is_none()
+            && seller_fee_basis_points.is_none()
+            && metadata_json.is_none()
+        {
+            return Err(Error::new("please select atleast one field to update"));
+        }
+
+        let AppContext { db, user_id, .. } = ctx.data::<AppContext>()?;
+        let UserID(id) = user_id;
+        let conn = db.get();
+        let producer = ctx.data::<Producer<NftEvents>>()?;
+        let nft_storage = ctx.data::<NftStorageClient>()?;
+        let solana = ctx.data::<Solana>()?;
+
+        let user_id = id.ok_or_else(|| Error::new("X-USER-ID header not found"))?;
+
+        let (drop_model, solana_collection_model) = Drops::find()
```

Comment:
Have to change this to make it generic for all blockchains. Plan is to find drop and collection -> match the blockchain

Reply:
Should just do an extra query in the update method of the solana.editions().update() to find the required addresses and update the record.